cmd/clusterctl/client/cluster/proxy.go

```diff
@@ -89,8 +95,35 @@ func (k *proxy) ValidateKubernetesVersion() error {
 	return nil
 }
 
+func (k *proxy) GetConfig() (*rest.Config, error) {
+	config, err := clientcmd.LoadFromFile(k.kubeconfigPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to load Kubeconfig file from %q", k.kubeconfigPath)
+	}
+
+	restConfig, err := clientcmd.NewNonInteractiveClientConfig(
```

Comment:
Is there a reason to switch to `NewNonInteractiveClientConfig` instead of  using `NewDefaultClientConfig` as before and just set the `Contect` field in the `ConfigOverrides{}` struct?

Reply:
Aside note. Should we set the timeout on the connection as well? If I'm not wrong no timeout means infinite, and this can lead to weird errors
@vincepri opinions ^^

Reply:
I'm fine either way, I think when @wfernandes and I looked at it, it wasn't clear you could override the configuration to set the context

Reply:
Whoa! Didn't see that in the `ConfigOverrides` struct. When I saw the definition of `NewDefaultClientConfig` my eyes were immediately drawn to the function argument. Nice...I'll try it with the overrides and make sure everything works as expected.

Reply:
@fabriziopandini I forgot about your comment regarding timeout. I'll add support to configure that and put in a default timeout. I'll set the default to be 30s unless we have a consistent timeout set for api-server requests in other places.